### PR TITLE
Add @ivp for system typed arg

### LINKED ISSUE #199: IVP macro with system
```julia
S = @system(x'=-x)
@ivp(S, x(0) ∈ X0)
```

idea:

```julia
macro ivp(expr...)
    try
        if isa(expr, Tuple{Symbol, Expr}) # handle @ivp(S::AbstractSystem, x(0) ∈ X0)
            T = ... # get type of expr[1]
            if isa(T, AbstractSystem)
                sys = expr[1]
                @capture(expr[2], x_(0) ∈ x0_) || throw(ArgumentError("malformed epxpression")) # TODO handle equality
                ivp = Expr(:call, InitialValueProblem, :($(expr[1])), :($x0))
                return esc(ivp)
            else
                error("...")
            end
        end
        dyn_eq, AT, constr, state, input, noise, dim, x0 = _parse_system(expr)
.....
```

### PR DESCRIPTION
**Pull request by ueliwechsler**
Closes #199.

### CODE REVIEW

**Comment by ueliwechsler** [comment]
sorry, I also had to add an extension to the docstring

**Comment by mforets** [comment]
saw it, looks good :+1:

File: src/macros.jl [hunk 1]
```diff
@@ -926,19 +929,32 @@ InitialValueProblem{LinearContinuousSystem{Float64,IdentityMultiple{Float64}},Ar
...
     try
-        dyn_eq, AT, constr, state, input, noise, dim, x0 = _parse_system(expr)
-        sys_type, var_names = _get_system_type(dyn_eq, AT, constr, state, input, noise, dim)
-        sys = Expr(:call, :($sys_type), :($(var_names...)))
-        if x0 == nothing
-            return throw(ArgumentError("an initial-value problem should define the " *
-                        "initial states, but such expression was not found"))
-        else
-            ivp = Expr(:call, InitialValueProblem, :($sys), :($x0))
+        # if the @ivp macro is called with a AbstractSystem typed arguement
```
Comment (schillic) [inline src/macros.jl:941]:
```suggestion
        # check if the @ivp macro is called with an AbstractSystem argument
```

File: src/macros.jl [hunk 2]
```diff
@@ -948,3 +964,16 @@ macro ivp(expr...)
         end
     end
 end
+
+function parses_as_system(expr)
+    # The argument is a system if
+    # it is directly defined using @system in the macro call
```
Comment (schillic) [inline src/macros.jl:970]:
```suggestion
    # 1) it is directly defined using @system in the macro call
```

File: src/macros.jl [hunk 3]
```diff
@@ -948,3 +964,16 @@ macro ivp(expr...)
...
+    if @capture(expr, @system(def_))
+        return true
+    end
+    # if it is a variable, i.e. it is not a dynamic equation
```
Comment (schillic) [inline src/macros.jl:974]:
```suggestion
    # 2) it is a variable, i.e. it is not a dynamic equation
```

**Pull request merged**